presto-orc/src/main/java/com/facebook/presto/orc/stream/OrcInputStream.java

```diff
@@ -310,6 +316,32 @@ private void advance()
         }
     }
 
+    @Override
+    public boolean hasNext()
+    {
+        return buffer == null;
+    }
+
+    @Override
+    public Buffer next()
+    {
+        if (buffer == null) {
+            throw new NoSuchElementException();
+        }
+        // advance may set bufferContainer to null
```

Comment:
@elonazoulay This comment is confusing to me. If bufferContainer becomes null, then the following code will throw NPE.

Reply:
Thanks, I'll clarify this and push the update shortly.